drakrun/drakrun/analyzer.py

```diff
@@ -0,0 +1,515 @@
+
+    time_started = time.time()
+    with run_vm(vm, options) as vm:
+        log.info("Copying sample to VM...")
+        target_path = drop_sample_to_vm(vm, options.sample_path, file_name)
+
+        if user_start_command:
+            start_command = user_start_command.replace("%f", target_path)
+        else:
+            start_command = get_sample_startup_command(
+                target_path, extension, sample_entrypoints
+            )
+        log.info("Using command: %s", start_command)
+
+        run_post_restore_vm_commands(vm, options)
+
+        cwd = ntpath.dirname(target_path)
+        drakvuf_cmd = build_drakvuf_cmdline(
+            vm, cwd, start_command, plugins=options.plugins, options=options
+        )
+
+        drakmon_log_path = output_dir / "drakmon.log"
+        with run_tcpdump(vm, options), drakmon_log_path.open("wb") as drakmon_log:
+            try:
+                subprocess.run(
+                    drakvuf_cmd,
+                    stdout=drakmon_log,
+                    check=True,
+                    timeout=options.timeout + 60,
+                )
+            except subprocess.CalledProcessError as e:
+                # see DRAKVUF src/exitcodes.h for more details
+                INJECTION_UNSUCCESSFUL = 4
+
+                if e.returncode == INJECTION_UNSUCCESSFUL:
+                    log.error("Sample startup failed")
+                else:
+                    # Something bad happened
+                    raise e
+            except subprocess.TimeoutExpired as e:
+                log.error("DRAKVUF timeout expired")
+                raise e
+
+        if options.raw_memory_dump:
+            vm.vm.memory_dump(str(output_dir / "post_sample.raw_memdump.gz"))
+
+    log.info("Analysis done. Collecting artifacts...")
+
+    # Make sure dumps have a reasonable size.
+    # Calculate dumps_metadata as it's required by the `analysis` task format.
+    dumps_path = output_dir / "dumps"
+    dumps_zip_path = output_dir / "dumps.zip"
+    dumps_metadata = crop_dumps(str(dumps_path), str(dumps_zip_path))
+
+    # Compress IPT traces, they're quite large however they compress well
+    ipt_path = output_dir / "ipt"
+    ipt_zip_path = output_dir / "ipt.zip"
+    compress_ipt(str(ipt_path), str(ipt_zip_path))
+
+    time_finished = time.time()
+    return {
+        "time_started": time_started,
+        "time_finished": time_finished,
+        "start_command": start_command,
+        "dumps_metadata": dumps_metadata,
+    }
+
+
+class PluginsArgAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs is not None:
+            raise ValueError("nargs not allowed")
+        super().__init__(option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if option_string is None:
+            raise ValueError("Missing --plugins value")
```

Comment:
Connection between the error message and the code is not obvious to me. Maybe add some docstring to the class to clear up what it's doing?